libraries/commerce/product/selectors/product.js

```diff
@@ -184,21 +184,68 @@ export const getProduct = createSelector(
   }
 );
 
+/**
+ * Retrieves the product name from the current route.
+ * @param {Object} state The current application state.
+ * @param {Object} props The component props.
+ * @return {string|null}
+ */
+export const getProductRouteName = createSelector(
+  getCurrentRoute,
+  (route) => {
+    if (!route || !route.state || !route.state.title) {
+      return null;
+    }
+
+    return route.state.title;
+  }
+);
+
 /**
  * Retrieves the product name.
  * @param {Object} state The current application state.
  * @param {Object} props The component props.
  * @return {string|null}
  */
 export const getProductName = createSelector(
-  getCurrentRoute,
+  getProductRouteName,
+  getProduct,
+  (routeName, product) => {
+    if (routeName) {
```

Comment:
I guess taking a title from a route should be a fallback for productData.title. It is possible that we refresh product data on PDP. We receive a new name, but still show an old one, because of route state.

Reply:
That is true. I don't even remember why we introduced this.